shared-libs/message-utils/src/index.js

```diff
@@ -96,21 +96,52 @@ const applyPhoneFilters = function(config, phone) {
   return phone;
 };
 
+const normalizeRecipient= function(recipient) {
+  if (Array.isArray(recipient)) {
+    return recipient.map(r => typeof r === 'string' ? r.trim() : r);
+  }
+
+  return typeof recipient === 'string'
+    ? recipient.trim()
+    : recipient;
+};
+
 const getRecipient = function(context, recipient, defaultToSender = true) {
   if (!context) {
     return;
   }
 
   const from = context.from || context.contact?.phone;
-  recipient = recipient?.trim();
+  recipient = normalizeRecipient(recipient);
 
   if (!recipient) {
```

Comment:
This needs be (!recipient.length), granted that `normalizeRecipient` will return an empty array when the provided recipient cannot be normalized.